sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/QueryTestRunner.java

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.benchmark.linkedin;
+
+import com.azure.cosmos.CosmosAsyncClient;
+import com.azure.cosmos.benchmark.Configuration;
+import com.azure.cosmos.benchmark.linkedin.data.EntityConfiguration;
+import com.azure.cosmos.benchmark.linkedin.data.Key;
+import com.azure.cosmos.benchmark.linkedin.impl.Constants;
+import com.azure.cosmos.benchmark.linkedin.impl.exceptions.AccessorException;
+import com.azure.cosmos.benchmark.linkedin.impl.models.QueryOptions;
+import com.azure.cosmos.models.SqlParameter;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.collect.ImmutableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The test implementation for GETs/pointed lookups against CosmosDB. It takes in
+ * the testData and submits upto N get operations at a time, leveraging the ExecutorService
+ * for the parallelism control.
+ *
+ * The Latency tracking and reposting is handled within the Accessor, and not in this method.
+ */
+public class QueryTestRunner extends TestRunner {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueryTestRunner.class);
+    private static final String PARTITIONING_KEY_BINDING_PARAMETER = "@" + Constants.PARTITION_KEY;
+    private static final String QUERY = String.format("SELECT * FROM c WHERE c.%s = %s",
```

Comment:
Can you also try having the query CTL run for queries with partition key in CosmosQueryOptions?
We made an optimization around simple single partition queries where setting the environment variable COSMOS.QUERYPLAN_CACHING_ENABLED to true would improve the performance of the query by avoiding the query plan calls to gateway. Let me know if you need more context on this. Thanks

Reply:
@simplynaveen20 had brought this item in a previous PR. Based on our discussion, I had added this: https://github.com/Azure/azure-sdk-for-java/blob/7786a3929bf09ac9f3f18dd67c2272bc0ae24007/sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/impl/QueryExecutor.java#L78 where we are explicitly specify the partitioningKey in the CosmosQueryOptions. Does that address your suggestion?

I have a question on the environment variable. How does one set that? I was wondering if there could be a configuration/client build time value that could be used to enable it?

Reply:
Yes thats exactly what I was suggesting too. So that is good.
You can set the environment variable in the CTL environment during our run if there is a special way of doing that @simplynaveen20  might know.
You can also set that directly in the code like this         System.setProperty("COSMOS.QUERYPLAN_CACHING_ENABLED", "true");

Reply:
@mbhaskar: Making it a property that can be set during the CosmosClientBuilder initialization would be a better way? Setting environment variables, though possible, separates the 'implementation logic' across the java code and the startup/initialization phase?

Reply:
@amarathavale  Right now this is an experimental enhancement. We would make it default behaviour based on feedback in which case you wouldnt even need an API for this. Thats why this is an environment variable at the moment and floating it to select users